lib/uniris/transaction_chain/transaction.ex

```diff
@@ -171,6 +173,13 @@ defmodule Uniris.TransactionChain.Transaction do
     {previous_public_key, next_public_key}
   end
 
+  defp get_transaction_public_keys(:oracle) do
```

Comment:
I think we should leverage a new derivation algorithm inspired from the beacon chain, to derive key based on the oracle and the time of the transaction. So we will have a cryptographic chain for the specific oracle.